### ISSUE TITLE
Uppercase alias ? (can't set "C#" alias for class="language-C#")

### ISSUE THREAD

**Issue by softlion** [highlightjs/highlight.js#3967]
I have that html code imported into my html from github:

```html
<code class="language-C#">
 ...
</code>

and that local html:
...
    <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.9.0/highlight.min.js"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.9.0/languages/csharp.min.js"></script>
    <script>
    hljs.registerAliases('C#', { languageName: 'csharp' });
    hljs.highlightAll();
    </script>
```

It looks like the C# alias can't be used in uppercase, The source code of `highlightAll` shows that is makes all alias lower case before adding them to the alias collection.

The result is, C# code is not detected. On the rendered HTML I see the hljs class and the hljs-c class.

Is there a workaround ?

**Comment by joshgoebel** [comment]
A few notes:

- by convention, aliases must be lowercase (to avoid ambiguity)
- also, `language-C#` is not a valid CSS class name either

---

But this is trivial to implement... if you want to do this just write your own version of `highlightAll` wrapper and use data tags instead of class names:

```
<code data-language="C#">
 ...
</code>
```

Of course you'll need your own mapping table/function to convert `C#` into `csharp`... so your function loops, converts your "pretty names" back to actual aliases, then calls `highlightElement`... done.

**Comment by Flayms** [comment]
I just stumbled into a similar / same problem. Here it doesn't have much to do with uppercase.
even
```html
<code class="language-c#">
 ...
</code>
```

gets converted into
```html
<code class="language-c# hljs language-c" data-highlighted="yes">
...
</code>
```
Registering 'c#' or 'language-c#' neither solves the problem.

I went a bit through the code and 'language-* takes precedence over non-prefixed class names'.
The regex which is designed for that:
```js
languageDetectRe: /\blang(?:uage)?-([\w-]+)\b/i
```
 uses \b which breaks at the # of c# and thus detects language-c instead.
This could be resolved by changing the reg into smth like
```js
/(?<=^|\s)lang(?:uage)?-([\w#-]+)(?=\s|$)/i
```
which would account for the # and also detect word bounds differently.
